src/Core/SettingsChangesHistory.cpp

```diff
@@ -82,6 +82,7 @@ const VersionToSettingsChangesMap & getSettingsChangesHistory()
             {"optimize_exchanges", 0, 0, "New experimental setting."},
             {"force_exchange_kind", "", "", "New experimental setting."},
             {"allow_experimental_delta_kernel_rs", false, false, "New setting"},
+            {"enable_shared_storage_snapshot_in_query", false, false, "A new setting to shard storage snapshot in query"},
```

Comment:
Egh, we will likely need to add `merge_tree_storage_snapshot_sleep_ms` here as well?